test/e2e/specs/test.spec.js

```diff
@@ -45,15 +95,37 @@ describe('Vaults UI Test Cases', () => {
         cy.contains(
           'p',
           'You can manage your vaults from the "My Vaults" view.',
+          { timeout: MINUTE_MS },
         ).should('exist');
+        cy.contains('Manage my Vaults').click();
       });
     });
 
-    it('should adjust the collateral by performing a withdrawl and approve the transaction successfully', () => {
-      cy.contains('button', 'Manage my Vaults').click();
-      cy.contains('button', 'Back to vaults').click();
-      cy.contains('div', /ATOM.*#8/).click();
+    it('should open the new vault', () => {
+      cy.get('span')
+        .contains(/My Vaults.*\(\d+\)/)
+        .children()
+        .first()
+        .spread(element => {
+          // Get the total number of vaults present
+          const vaultCountRegex = element.innerHTML.match(/\((\d+)\)/);
+          const vaultCount = Number(vaultCountRegex[1]);
+
+          cy.findAllByText(/#\d+/)
+            .should('have.length', vaultCount)
+            .spread((...vaults) => {
+              expect(
```

Comment:
I think we can use `.then` over here as well. `.then(vaults => {` would suffice

Reply:
spread is required here because then only passes the first matching elements as an argument